File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/service/MessagingService.java
package technology.mainthread.apps.gatekeeper.service;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import javax.inject.Inject;

import technology.mainthread.apps.gatekeeper.GatekeeperApp;
import technology.mainthread.apps.gatekeeper.view.NotifierHelper;
import timber.log.Timber;

public class MessagingService extends FirebaseMessagingService {

    @Inject
    NotifierHelper notifierHelper;

    @Override
    public void onCreate() {
        super.onCreate();
        GatekeeperApp.get(this).inject(this);
    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        Timber.d("From: %s", remoteMessage.getFrom());
        for (String key : remoteMessage.getData().keySet()) {
            Timber.d("%1$s is a key in the bundle, %2$s is the value", key, remoteMessage.getData().get(key));
        }

        notifierHelper.notifyHandsetCalling();
    }
}

File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/service/MessagingService.kt
package technology.mainthread.apps.gatekeeper.service

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import dagger.android.AndroidInjection
import technology.mainthread.apps.gatekeeper.view.notificaton.NotifierHelper
import timber.log.Timber
import javax.inject.Inject

class MessagingService : FirebaseMessagingService() {

    @Inject
    internal lateinit var notifierHelper: NotifierHelper

    override fun onCreate() {
        AndroidInjection.inject(this)
        super.onCreate()
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage?) {
        Timber.d("From: %s", remoteMessage!!.from)
        for (key in remoteMessage.data.keys) {
            Timber.d("%1\$s is a key in the bundle, %2\$s is the value", key, remoteMessage.data[key])
        }

        notifierHelper.handlePushNotification(remoteMessage.from)
    }
}
